operator/gitops/argocd/pipeline-service/openshift-pipelines/chains-secrets-config.yaml

```
@@ -61,37 +61,43 @@ spec:
             - /bin/bash
             - -c
             - |
+              set -o errexit
+              set -o nounset
+              set -o pipefail
+
+              namespace="openshift-pipelines"
+              secret="signing-secrets"
+
               cd /tmp
-              # Try to handle that nicely. The object is expected to always exist so check the data.
-              SIG_KEY_DATA=$(kubectl get secret signing-secrets -n openshift-pipelines -o jsonpath='{.data}')
-              if [[ -n $SIG_KEY_DATA ]]; then
+
+              if [ "$(kubectl get secret "$secret" -n "$namespace" -o jsonpath='{.data}')" != "" ]; then
```

This will not exit with non zero and output an empty string when the secret is missing or the `data` key is empty

```suggestion
              if [ "$(kubectl get secret "$secret" -n "$namespace" -o jsonpath='{.data}' --ignore-not-found --allow-missing-template-keys)" != "" ]; then
```

---

Because it is in an `if` statement, the failure was not triggering the script to exit. This is confirmed by the tests being successful.

I will still add the options because it does not harm either and is more explicit.